pkg/terraformer/terraformer_test.go

```diff
@@ -99,6 +99,15 @@ var _ = Describe("Terraformer", func() {
 				Expect(testClient.Delete(ctx, testObjs.ConfigurationConfigMap)).To(Succeed())
 				Expect(tf.Run(terraformer.Apply)).To(MatchError(ContainSubstring("not found")))
 			})
+			It("should succeed because state is empty", func() {
+				emptyState := testObjs.StateConfigMap
+				emptyState.Data = nil
```

Comment:
Without the finalizer to be set in advance, the test will be always successful because there is nothing to add it.
```suggestion
				emptyState.Data = nil
				emptyState.SetFinalizers([]string{terraformer.TerraformerFinalizer})
```

Please, add a test case where a different data key in the configmap is used.